Maple2Storage/Types/Metadata/AdditionalEffectMetadata.cs

```diff
@@ -0,0 +1,262 @@
+﻿using System.Xml.Serialization;
+using Maple2Storage.Enums;
+using MapleServer2.Enums;
+
+namespace Maple2Storage.Types.Metadata;
+
+[XmlType]
+public class AdditionalEffectMetadata
+{
+    [XmlElement(Order = 1)]
+    public int Id;
+    [XmlElement(Order = 2)]
+    public Dictionary<int, List<AdditionalEffectLevelMetadata>> Levels;
+
+    public AdditionalEffectLevelMetadata GetLevel(int level, string feature = "")
+    {
+        List<AdditionalEffectLevelMetadata> levels;
+        Levels.TryGetValue(level, out levels);
+
+        if (levels == null)
+        {
+            return null;
+        }
+
+        if (feature == "")
+        {
+            return levels[0];
+        }
+
+        foreach (AdditionalEffectLevelMetadata levelMeta in levels)
+        {
+            if (levelMeta.Feature == feature)
+            {
+                return levelMeta;
+            }
+        }
+
+        return null;
+    }
+}
```

Comment:
We usually dont put functions inside the metadata files and they should go in the storage file